### PR TITLE
Sort APIResources into a consistent order

### PR DESCRIPTION
**Pull request by justinsb**
Should ensure more predictable test results.

### CODE REVIEW

**Comment by justinsb** [comment]
/retest

I think these tests are going to be flaky till we get the watches in BeforeApply (https://github.com/kubernetes-sigs/kubebuilder-declarative-pattern/pull/277#issuecomment-1451330431)

File: mockkubeapiserver/apiresourcelist.go [hunk 1]
```diff
@@ -47,5 +48,19 @@ func (req *apiResourceList) Run(ctx context.Context, s *MockKubeAPIServer) error
 		}
 		response.APIResources = append(response.APIResources, resource)
 	}
+
+	// Return in a stable order, for more test predictability
```
Comment (yuwenma) [inline mockkubeapiserver/apiresourcelist.go:52]: Does this sorting for consistent output diff check?

Comment (justinsb) [reply mockkubeapiserver/apiresourcelist.go:52]: Right :-)  I don't think it matters to real clients

**Comment by justinsb** [comment]
Sigh, this is the flake we're trying to fix...

/retest

**Comment by justinsb** [comment]
Hooray - fourth time's the charm :joy:

**Pull request merged**